src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs

```diff
@@ -57,57 +54,71 @@ public static void AreEqual(FileInfo expected, Rectangle expectedRect, FileInfo
+					.AppendLine("actual  : " + actual?.Name + (actualRect == FirstQuadrant ? null : $"in {actualRect}"))
+					.AppendLine("====================")
+					.ApplyIf(message != null, x => x.AppendLine(message))
+					.ApplyIf(builder != null, builder)
+					.ToString();
+			}
 		}
+		#endregion
 
+		#region AreAlmostEqual
 		/// <summary>
 		/// Asserts that two screenshots are equal to each other inside the area of the nominated rect to within the given pixel error
 		/// (ie, the A, R, G, or B values cumulatively differ by more than the permitted error for any pixel).
 		/// </summary>
-		public static void AreAlmostEqual(FileInfo expected, FileInfo actual, IAppRect rect, int permittedPixelError)
-			=> AreAlmostEqual(expected, actual, new Rectangle((int)rect.X, (int)rect.Y, (int)rect.Width, (int)rect.Height), permittedPixelError);
+		public static void AreAlmostEqual(FileInfo expected, FileInfo actual, IAppRect rect, int permittedPixelError, [CallerLineNumber] int line = 0)
+			=> AreAlmostEqualImpl(expected, rect.ToRectangle(), actual, rect.ToRectangle(), permittedPixelError, line);
 
 		/// <summary>
 		/// Asserts that two screenshots are equal to each other inside the area of the nominated rect to within the given pixel error
 		/// (ie, the A, R, G, or B values cumulatively differ by more than the permitted error for any pixel).
 		/// </summary>
-		public static void AreAlmostEqual(FileInfo expected, FileInfo actual, Rectangle? rect = null, int permittedPixelError = 5)
-		{
-			rect = rect ?? new Rectangle(0, 0, int.MaxValue, int.MaxValue);
-			AreAlmostEqual(expected, rect.Value, actual, rect.Value, permittedPixelError);
-		}
+		public static void AreAlmostEqual(FileInfo expected, FileInfo actual, Rectangle? rect = null, int permittedPixelError = 5, [CallerLineNumber] int line = 0)
+			=> AreAlmostEqualImpl(expected, rect ?? FirstQuadrant, actual, rect ?? FirstQuadrant, permittedPixelError, line);
 
 		/// <summary>
 		/// Asserts that two screenshots are equal to each other inside the area of the nominated rect to within the given pixel error
 		/// (ie, the A, R, G, or B values cumulatively differ by more than the permitted error for any pixel).
 		/// </summary>
-		public static void AreAlmostEqual(FileInfo expected, IAppRect expectedRect, FileInfo actual, IAppRect actualRect, int permittedPixelError)
-			=> AreAlmostEqual(
-				expected,
-				new Rectangle((int)expectedRect.X, (int)expectedRect.Y, (int)expectedRect.Width, (int)expectedRect.Height),
-				actual,
-				new Rectangle((int)actualRect.X, (int)actualRect.Y, (int)actualRect.Width, (int)actualRect.Height),
-				permittedPixelError);
+		public static void AreAlmostEqual(FileInfo expected, IAppRect expectedRect, FileInfo actual, IAppRect actualRect, int permittedPixelError, [CallerLineNumber] int line = 0)
+			=> AreAlmostEqualImpl(expected, expectedRect.ToRectangle(), actual, actualRect.ToRectangle(), permittedPixelError, line);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'actualRect' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5170992)